components/Releases/Releases.jsx

```diff
@@ -13,12 +13,35 @@ export default function Releases({ profileData }) {
   const [releases, setReleases] = useState([])
   const [allowNew, setAllowNew] = useState(true)
   const [addedNewRelease, setAddedNewRelease] = useState(false)
+  const [filter, setFilter] = useState("oldest")
 
   useEffect(() => {
     getReleases()
     setAddedNewRelease(false)
   }, [supabase, profileData.id, addedNewRelease])
 
+  useEffect(() => {
+    if (releases) {
+      let sortedReleases = releases
+      if (filter === "alphabetical") {
+        sortedReleases.sort((a, b) => {
+          let titleOne = a.title.toLowerCase()
+          let titleTwo = b.title.toLowerCase()
+
+          if (titleOne < titleTwo) {
+            return -1
+          }
+          if (titleOne > titleTwo) {
+            return 1
+          }
+          return 0
+        })
+      }
+
+      setReleases(sortedReleases)
+    }
+  }, [filter])
```

Comment:
Let's remove this and instead create a function. This function can reference values in a `select` menu. These sorts are very crude but I believe they should work. There may be some other things to check here though.

```jsx
const handleSort = (value) => {
  let arr;

  if (value === "created") {
    arr = [...releases].sort((a, b) => a.created_at > b.created_at ? 1 : -1);
  }
  if (value === "ascending") {
    arr = [...releases].sort((a, b) => a.title > b.title ? 1 : -1);
  }

  setReleases(arr)
}
```

And the HTML to replace your button:

```jsx
<select onChange={(e) => handleSort(e.target.value)}>
  <option value="created" selected>Created</option>
  <option value="ascending">Ascending</option>
</select>
```

Reply:
I have done this and expanded the options. Works perfect. Also implemented on user profile page.